src/Microsoft.IdentityModel.Tokens/TokenUtilities.cs

```diff
@@ -55,5 +58,70 @@ internal static IEnumerable<SecurityKey> GetAllSigningKeys(TokenValidationParame
                 foreach (SecurityKey key in validationParameters.IssuerSigningKeys)
                     yield return key;
         }
+
+        /// <summary>
+        /// Merges claims. If a claim with same type exists in both the lists, the one in claims will be kept.
+        /// </summary>
+        /// <param name="claims"> Collection of claims.</param>
+        /// <param name="subjectClaims"> Collection of claims.</param>
+        /// <returns> A Merged list of claims.</returns>
+        internal static IEnumerable<Claim> MergeClaims(IEnumerable<Claim> claims, IEnumerable<Claim> subjectClaims)
+        {
+            if (claims == null)
+                return subjectClaims;
+
+            if (subjectClaims == null)
+                return claims;
+
+            List<Claim> result = claims.ToList();
+
+            foreach (Claim claim in subjectClaims)
+            {
+                if (claims.Where(i => i.Type == claim.Type).FirstOrDefault() == null)
+                    result.Add(claim);
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Gets the value type of the claim.
+        /// </summary>
+        /// <param name="value"> The claim value.</param>
+        /// <returns> The value type of the claim.</returns>
+        internal static string GetClaimValueTypeFromValue(object value)
+        {
+            if (value != null)
+            {
+                if (value.GetType().Name == typeof(String).Name)
```

Comment:
Prefer language keywords instead of BCL types

See item 11 here https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/coding-style.md

Reply:
Good to know.